pipeline/extract/chat.py

```diff
@@ -0,0 +1,113 @@
+from typing import Dict, List, Set
+
+import pandas as pd
+import prefect
+from prefect import task
+
+from pipeline.extract.users import extract_users
+from pipeline.transform.users import convert_users_into_hash
+from pipeline.types import (
+    ChatData,
+    ChatMatchId,
+    Community,
+    Match,
+    MatchMetadata,
+    Message,
+    User,
+    UserId,
+)
+
+
+@task
+def extract_recent_chatdata(db, community: Community) -> List[ChatData]:
+    logger = prefect.context.get("logger")
+    chat_ref = db.reference(f"matches/{community}")
+    user_chat_records = chat_ref.get()
+    if not user_chat_records:
+        logger.info(f"No user-chat records for commmnity: {community}")
+        return []
+    logger.info(f"Extracted {len(user_chat_records)} user-chat records.")
+    ids: Set[ChatMatchId] = set()
+    chatList: List[ChatData] = []
+
+    df_users = extract_users.run(db, community)
+    users = convert_users_into_hash(df_users)
+
+    for key in user_chat_records.keys():
+        if user_chat_records[key]["id"] not in ids:
```

Comment:
It would be more readable to invert this condition and continue the loop.

Instead of:

```python
for blah in blahs:
    if condition:
        do_thing_1()
        do_thing_2()
        do_thing_3()
```

Do this:

```python
for blah in blahs:
    if not condition:
        continue
    do_thing_1()
    do_thing_2()
    do_thing_3()
```

Check the opposite of the condition and continue to the next iteration of the loop.

This makes it easier to follow the loop logic linearly. It also avoids having deeply-indented code blocks, which become harder to read and maintain.